panic: runtime error: slice bounds out of range

```
ligi@komputing:~$ ipfs daemon
Initializing daemon...
go-ipfs version: 0.4.21-
Repo version: 7
System version: amd64/linux
Golang version: go1.12.5
Swarm listening on /ip4/10.10.42.1/tcp/4001
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/172.17.0.1/tcp/4001
Swarm listening on /ip4/178.19.221.38/tcp/4001
Swarm listening on /ip6/::1/tcp/4001
Swarm listening on /p2p-circuit
Swarm announcing /ip4/10.10.42.1/tcp/4001
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/172.17.0.1/tcp/4001
Swarm announcing /ip4/178.19.221.38/tcp/4001
Swarm announcing /ip6/::1/tcp/4001
API server listening on /ip4/0.0.0.0/tcp/5001
WebUI: http://0.0.0.0:5001/webui
Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready
panic: runtime error: slice bounds out of range

goroutine 1510682702 [running]:
github.com/multiformats/go-multiaddr-dns.matchDnsaddr(0x1ac4400, 0xc0113d6a80, 0xc037240710, 0x4, 0x7, 0x0)
        pkg/mod/github.com/multiformats/go-multiaddr-dns@v0.0.2/resolve.go:177 +0x114
github.com/multiformats/go-multiaddr-dns.(*Resolver).resolveDnsaddr(0x237b7f0, 0x1aadb00, 0xc03920d4c0, 0x1ac4400, 0xc00a59ece0, 0x61, 0x86, 0xc029207180, 0xc0460f0af0, 0xc076dd4090)
        pkg/mod/github.com/multiformats/go-multiaddr-dns@v0.0.2/resolve.go:167 +0x4d6
github.com/multiformats/go-multiaddr-dns.(*Resolver).Resolve(0x237b7f0, 0x1aadb00, 0xc03920d4c0, 0x1ac4400, 0xc00a59ece0, 0xc037240680, 0x4, 0x8, 0x0, 0x0)
        pkg/mod/github.com/multiformats/go-multiaddr-dns@v0.0.2/resolve.go:79 +0xd3
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).resolveAddrs(0xc0001a7ad0, 0x1aadb00, 0xc03920d4c0, 0xc0353ffa40, 0x22, 0xc01f91aea0, 0x11, 0x11, 0xb042a0, 0xc00038c200, ...)
        pkg/mod/github.com/libp2p/go-libp2p@v0.0.28/p2p/host/basic/basic_host.go:505 +0x36f
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).Connect(0xc0001a7ad0, 0x1aadb00, 0xc03920d4c0, 0xc0353ffa40, 0x22, 0x0, 0x0, 0x0, 0x1a831c0, 0xc0746efbc0)
        pkg/mod/github.com/libp2p/go-libp2p@v0.0.28/p2p/host/basic/basic_host.go:481 +0x1b3
github.com/libp2p/go-libp2p-kad-dht.(*dhtQueryRunner).dialPeer(0xc04313d560, 0x1aadb00, 0xc03920d4c0, 0xc0353ffa40, 0x22, 0xb1e0e7c, 0x132b1ed1600094)
        pkg/mod/github.com/libp2p/go-libp2p-kad-dht@v0.0.13/query.go:251 +0x1df
github.com/libp2p/go-libp2p-kad-dht.(*dialQueue).worker(0xc021c759a0)
        pkg/mod/github.com/libp2p/go-libp2p-kad-dht@v0.0.13/dial_queue.go:327 +0x384
created by github.com/libp2p/go-libp2p-kad-dht.(*dialQueue).control
        pkg/mod/github.com/libp2p/go-libp2p-kad-dht@v0.0.13/dial_queue.go:141 +0xd5
```

daemon was running for a while 

```
ligi@komputing:~$ ipfs version --all
go-ipfs version: 0.4.21-
Repo version: 7
System version: amd64/linux
Golang version: go1.12.5
```

--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,6 @@ require (
 	github.com/Kubuxu/gocovmerge v0.0.0-20161216165753-7ecaa51963cd
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/bren2010/proquint v0.0.0-20160323162903-38337c27106d
-	github.com/cenkalti/backoff v2.1.1+incompatible
 	github.com/dustin/go-humanize v1.0.0
 	github.com/elgris/jsondiff v0.0.0-20160530203242-765b5c24c302
 	github.com/fatih/color v1.7.0 // indirect
@@ -29,7 +28,6 @@ require (
 	github.com/ipfs/go-ds-measure v0.0.1
 	github.com/ipfs/go-fs-lock v0.0.1
 	github.com/ipfs/go-ipfs-blockstore v0.0.1
-	github.com/ipfs/go-ipfs-blocksutil v0.0.1
 	github.com/ipfs/go-ipfs-chunker v0.0.1
 	github.com/ipfs/go-ipfs-cmds v0.1.0
 	github.com/ipfs/go-ipfs-config v0.0.6
@@ -91,7 +89,7 @@ require (
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/mr-tron/base58 v1.1.2
 	github.com/multiformats/go-multiaddr v0.0.4
-	github.com/multiformats/go-multiaddr-dns v0.0.2
+	github.com/multiformats/go-multiaddr-dns v0.0.3
 	github.com/multiformats/go-multiaddr-net v0.0.1
 	github.com/multiformats/go-multibase v0.0.1
 	github.com/multiformats/go-multihash v0.0.5
@@ -112,7 +110,6 @@ require (
 	go4.org v0.0.0-20190313082347-94abd6928b1d // indirect
 	golang.org/x/sync v0.0.0-20190423024810-112230192c58 // indirect
 	golang.org/x/sys v0.0.0-20190626221950-04f50cda93cb
-	google.golang.org/appengine v1.4.0 // indirect
 	gopkg.in/cheggaaa/pb.v1 v1.0.28
 	gotest.tools/gotestsum v0.3.4
 )

